src/position.ts

```diff
@@ -470,66 +481,52 @@ export class UserPosition extends PositionWithRunner {
+
       collateralPermitSignature = await this.checkTokenAllowance(
         collateralTokenContract,
         userAddress,
         positionManagerAddress,
         new Decimal(absoluteCollateralChangeValue, Decimal.PRECISION),
-        TOKENS_WITH_PERMIT.has(collateralToken),
+        tokenConfig.supportsPermit,
         options,
       );
     }
 
-    switch (collateralToken) {
-      case 'ETH': // TODO - Update this case to use new ETH delegate once it's deployed
-        if (!isCollateralIncrease) {
-          throw new Error('ETH withdrawal from the position is not supported');
-        }
-
-        return sendTransactionWithGasLimit(
-          this.loadPositionManagerStETH().managePositionETH,
-          [absoluteDebtChangeValue, isDebtIncrease, maxFeePercentageValue, rPermitSignature],
-          gasLimitMultiplier,
-          frontendTag,
-          this.user,
+    if (isUnderlyingToken) {
+      return sendTransactionWithGasLimit(
+        this.positionManager.managePosition,
+        [
+          RaftConfig.getTokenAddress(collateralToken),
+          userAddress,
           absoluteCollateralChangeValue,
-        );
-
-      case 'stETH':
-        return sendTransactionWithGasLimit(
-          this.loadPositionManagerStETH().managePositionStETH,
-          [
-            absoluteCollateralChangeValue,
-            isCollateralIncrease,
-            absoluteDebtChangeValue,
-            isDebtIncrease,
-            maxFeePercentageValue,
-            rPermitSignature,
-          ],
-          gasLimitMultiplier,
-          frontendTag,
-          this.user,
-        );
-
-      case 'WETH':
-      case 'wstETH':
-        return sendTransactionWithGasLimit(
-          this.positionManager.managePosition,
-          [
-            RaftConfig.getTokenAddress(collateralToken),
-            userAddress,
-            absoluteCollateralChangeValue,
-            isCollateralIncrease,
-            absoluteDebtChangeValue,
-            isDebtIncrease,
-            maxFeePercentageValue,
-            collateralPermitSignature,
-          ],
-          gasLimitMultiplier,
-          frontendTag,
-          this.user,
-        );
+          isCollateralIncrease,
+          absoluteDebtChangeValue,
+          isDebtIncrease,
+          maxFeePercentageValue,
+          collateralPermitSignature,
+        ],
+        gasLimitMultiplier,
+        frontendTag,
+        this.user,
+      );
     }
+
+    const tokenConfig = RaftConfig.networkConfig.tokenTickerToTokenConfigMap[collateralToken];
+    return sendTransactionWithGasLimit(
+      this.loadDelegate(collateralToken)[tokenConfig.positionMangerManageFunc],
+      [
+        absoluteCollateralChangeValue,
+        isCollateralIncrease,
+        absoluteDebtChangeValue,
+        isDebtIncrease,
+        maxFeePercentageValue,
+        rPermitSignature,
+      ],
+      gasLimitMultiplier,
+      frontendTag,
+      this.user,
```

Comment:
Even if forget ETH deposit support for wstETH, this still won't work as it is missing `value`. For me, this sounds like a factory pattern where we create a transaction based on underlying and managing collateral tokens.

I have already tried to remove the `switch` without a factory and the main pain point is that ETH with wstETH has a completely different list of arguments compared to the others. When we add WETH, it'll become a mess since it has same args, but also needs a `value` since the position manager's function is a `payable`.